addons/crm/tests/test_crm_pls.py

```diff
@@ -593,6 +593,100 @@ def test_pls_no_share_stage(self):
+            self._prepare_test_lead_values(team_id, 'lead Won B', country_ids[1], state_ids[0], False, False, False, stage_ids[0]),
+            self._prepare_test_lead_values(team_id, 'lead Lost C', False, False, False, False, source_ids[0], stage_ids[0]),
+            self._prepare_test_lead_values(team_id, 'lead Lost D', country_ids[0], False, False, False, source_ids[0], stage_ids[0]),
+            self._prepare_test_lead_values(team_id, 'lead Lost E', False, state_ids[1], False, False, False, stage_ids[2]),
+            self._prepare_test_lead_values(team_id, 'lead Tooltip', country_ids[0], state_ids[0], False, False, source_ids[0], stage_ids[1]),
+        ])
+
+        # On creation, as phone and email are not set, these two fields will be set to False
+        leads.email_state = 'correct'
+        (leads[0] | leads[1] | leads[4] | leads[5]).phone_state = 'correct'
+        (leads[2] | leads[3]).phone_state = 'incorrect'
+
+        leads[:2].action_set_won()
+        leads[2:5].action_set_lost()
+        Lead._cron_update_automated_probabilities()
+        self.env.invalidate_all()
+
+        # Values for leads[5]:
+        # pW / pL is the probability that a won / lost lead has the lead value for a given field
+        # [Score = pW / (pW + pL)] -> A score above .5 is a TOP, below .5 a LOW, equal to .5 ignored
+        # Exception : for stage_id -> Score = 1 - P(current stage or lower for a lost lead)
+        # ------------------------------------------------------------------------------------------
+        # -- LOW 3 (lowest first, only 2 here)
+        # source_id:   pW = 0.1/1.2  pL = 2.1/2.2            -> Score = 0.08
+        # country_id:  pW = 1.1/2.2  pL = 1.1/1.2            -> Score = 0.353
+        # -- Neither
+        # email_state: pW = 2.1/2.1  pL = 3.1/3.1            -> Score = 0.5
+        # -- TOP 3 (highest first)
+        # state_id:    pW = 2.1/2.2  pL = 0.1/1.2            -> Score = 0.92
+        # phone_state: pW = 2.1/2.2  pL = 1.1/3.2            -> Score = 0.735
+        # stage_id:                  pL = 1.1/3.1            -> Score = 0.645
+        expected_low_3 = ['source_id', 'country_id']
+        expected_top_3 = ['state_id', 'phone_state', 'stage_id']
+
+        tooltip_data = leads[5].prepare_pls_tooltip_data()
+        self.assertEqual('Team Tooltip', tooltip_data['team_name'])
+        self.assertEqual(tools.float_compare(tooltip_data['probability'], 74.30, 2), 0)
+
+        self.assertEqual(len(tooltip_data['top_3_data']), len(expected_top_3))
+        self.assertEqual(len(tooltip_data['low_3_data']), len(expected_low_3))
+
+        for expected_top, data_top in zip(expected_top_3, tooltip_data['top_3_data']):
+            self.assertEqual(expected_top, data_top['field'])
+        for expected_low, data_low in zip(expected_low_3, tooltip_data['low_3_data']):
+            self.assertEqual(expected_low, data_low['field'])
```

Comment:
perhaps this (not tested)
```suggestion
        self.assertListEqual(list(map(lambda d: d['field'], tooltip_data['top_3_data'].values())), expected_top_3)
        self.assertListEqual(list(map(lambda d: d['field'], tooltip_data['low_3_data'].values())), expected_low_3)
```